wasm_builder_cli/src/main.rs

```diff
@@ -0,0 +1,108 @@
+use std::{
+    io::{stdout, BufWriter, Write},
+    path::PathBuf,
+};
+
+use clap::{Args, Parser};
+use color_eyre::eyre::Context;
+use iroha_wasm_builder::Builder;
+
+#[derive(Parser, Debug)]
+#[command(name = "iroha_wasm_builder_cli", version, author)]
+enum Cli {
+    /// Apply `cargo check` to the smartcontract
+    Check {
+        #[command(flatten)]
+        common: CommonArgs,
+    },
+    /// Build the smartcontract
+    Build {
+        #[command(flatten)]
+        common: CommonArgs,
+        /// Enable smartcontract formatting using `cargo fmt`.
+        // TODO: why it is a part of `build` in wasm_builder?
```

Comment:
Anyone who is in the context: could you provide a reasoning about this decision?

Reply:
Why have it an option as `build` and not a separate action, like `cargo build` and `cargo fmt`?

Reply:
Well it is a separate action that is run conditionally depending on whether this flag in the builder is set, isn't it?

Reply:
I'm puzzled about the inclusion of `cargo fmt` in the build process. While it makes sense for `wasm_builder` to handle `cargo check` and `cargo build` due to specific rustc parameters, `cargo fmt` doesn't seem to need such configuration. So why is it part of the build? Is formatting not the developer's responsibility? Are there historical reasons for this design choice?

Reply:
Yeah, tbh, it does feel kind of unnecessary, especially if `cargo fmt` does not need any special flags

Reply:
I'd suppose it is there to just make an all-inclusive package then. And maybe @Arjentix could clarify it more, but that's the gut feeling I have.

Reply:
In order to avoid piling up this PR, moved the discussion here: https://github.com/hyperledger/iroha/issues/3858